File: 4_1_10.jl
using Random

function random_numbers_and_count(n, a, b)
    numbers = rand(a:b, n)
    counts = Dict(num => count(x -> x == num, numbers) for num in unique(numbers))
    return numbers, counts
end

numbers, counts = random_numbers_and_count(10, 1, 5)
println("Numbers: ", numbers)
println("Counts: ", counts)

File: 4_1_10.cpp
#include <iostream>
#include <vector>
#include <map>
#include <random>
using namespace std;

pair<vector<int>, map<int, int>> randomNumbersAndCount(int n, int a, int b) {
    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<> dist(a, b);

    vector<int> numbers;
    map<int, int> counts;
    for (int i = 0; i < n; i++) {
        int num = dist(gen);
        numbers.push_back(num);
        counts[num]++;
    }
    return {numbers, counts};
}

int main() {
    auto [numbers, counts] = randomNumbersAndCount(10, 1, 5);

    cout << "Numbers: ";
    for (int num : numbers) cout << num << " ";
    cout << endl;

    cout << "Counts: ";
    for (auto& [num, count] : counts) {
        cout << num << ": " << count << " ";
    }
    cout << endl;
    return 0;
}
